Mage/src/main/java/mage/abilities/effects/common/DamageTargetEffect.java

```diff
@@ -152,13 +173,23 @@ public String getText(Mode mode) {
         StringBuilder sb = new StringBuilder();
         String message = amount.getMessage();
         sb.append(this.sourceName).append(" deals ");
-        if (message.isEmpty() || !message.equals("1")) {
+        if(whereXWording) {
+            sb.append("X ");
+        }
+        else if(OneDamageForEachWording) {
+            sb.append("1 ");
+        }
+        else if ((amount instanceof StaticValue) && !message.equals("1")) {
```

Comment:
Too much params for same thing (else/else/else). Try to use one param or add runtime check for correct usage (your current text gen un-support multiple usage of that params). One param example:
withAmountText("X")
withAmountText("1")
withAmountText("") // use amount value

Also I recommends to add examples for each text gen param like "param def; // 1 damage -> x damage".

`DamageTargetEffect` is good example of overwhelming of text gen params, it's very hard to use now, even after look at text gen code.

Reply:
Maybe @xenohedron can help with the ideas here

Reply:
Yeah I'll be back tomorrow and will take a look at these